src/raftstore/store/peer.rs

```diff
@@ -723,19 +773,77 @@ impl Peer {
         }
     }
 
-    /// Try to update lease.
-    ///
-    /// If the it can make sure that its lease is the latest lease, returns true.
-    fn maybe_update_lease(&mut self, term: u64, data: &[u8]) -> bool {
-        let mut req = RaftCmdRequest::new();
-        let propose_time = match req.merge_from_bytes(data)
-            .ok()
-            .and_then(|_| util::get_uuid_from_req(&req))
-            .and_then(|uuid| self.find_propose_time(uuid, term)) {
-            Some(t) => t,
-            _ => return false,
-        };
+    fn apply_reads(&mut self, ready: &Ready) {
+        let mut propose_time = None;
+        if self.ready_to_handle_read() {
+            for state in &ready.read_states {
+                let read = self.pending_reads.reads.pop_front().unwrap();
+                assert_eq!(state.request_ctx.as_slice(), read.uuid.as_bytes());
+                for (req, cb) in read.cmds {
+                    self.handle_read(req, cb);
+                }
+                propose_time = Some(read.renew_lease_time);
+            }
+        } else {
+            for state in &ready.read_states {
+                let read = &self.pending_reads.reads[self.pending_reads.ready_cnt];
+                assert_eq!(state.request_ctx.as_slice(), read.uuid.as_bytes());
+                self.pending_reads.ready_cnt += 1;
```

Comment:
It should record `state.index` for the read request, and check this `state.index` before performing read, after the log entry of this index is applied.